### PR TITLE
Fb sdk update

### PR DESCRIPTION
**Pull request by enginvardar**
Facebook SDK used for Facebook module was outdated and abandoned by Facebook, This PR updates updates the Facebook Module to the latest Facebook SDK and adds name option to test user to be created.

### CODE REVIEW

**Comment by enginvardar** [comment]
rerun travis

File: tests/unit/Codeception/Module/FacebookTest.php [hunk 1]
```diff
@@ -11,12 +11,13 @@
 class FacebookTest extends \PHPUnit_Framework_TestCase
 {
     protected $config = array(
-        'app_id'    => '460287924057084',
-        'secret'    => 'e27a5a07f9f07f52682d61dd69b716b5',
+        'app_id' => '1584417941835888',
+        'secret' => '17e014121a007dc5d149ada7f9201d07',
```
Comment (samdark) [inline tests/unit/Codeception/Module/FacebookTest.php]: Are you sure these should be changed? I guess not.

File: tests/unit/Codeception/Module/FacebookTest.php [hunk 2]
```diff
@@ -11,12 +11,13 @@
...
         'test_user' => array(
             'permissions' => array(
-                'publish_stream'
-            )
+                'public_profile',
```
Comment (samdark) [inline tests/unit/Codeception/Module/FacebookTest.php]: There's no need for any special permission for it. Public profile is there by default in v4 API.

**Comment by samdark** [comment]
Overall looks OK except application ID change and unused permission.

**Comment by enginvardar** [comment]
@samdark i've changed back application id and removed redundant permission, also i've updated documentation and find out that if test_user config does not exist test fail because of not found name key and fixed that too.

**Comment by samdark** [comment]
Great. Looks good to me now.

Will be checked once again by @DavertMik when he'll be back from vacation and then merged. Thanks!

File: src/Codeception/Lib/Driver/Facebook.php [hunk 3]
```diff
@@ -15,93 +18,98 @@ public function __construct($config, $logCallback = null)
...
 
     /**
-     * @inheritdoc
+     * @param $name
+     * @param array $permissions
+     *
+     * @return array
      */
-    protected function setPersistentData($key, $value)
+    public function createTestUser($name, array $permissions)
```
Comment (tiger-seo) [inline src/Codeception/Lib/Driver/Facebook.php]: name isn't required by Facebook, so i think it shouldn't be required by Codeception ;)

File: src/Codeception/Lib/Driver/Facebook.php [hunk 4]
```diff
@@ -15,93 +18,98 @@ public function __construct($config, $logCallback = null)
...
+    public function sendPostToFacebook($testUserAccessToken, array $parameters)
     {
-        // TODO: Implement clearAllPersistentData() method.
+        $response = $this->executeFacebookRequest(
+            new FacebookSession($testUserAccessToken),
+            'POST',
+            '/me/feed',
+            $parameters
+        )->getRawResponse();
+        return json_decode($response, true);
```
Comment (tiger-seo) [inline src/Codeception/Lib/Driver/Facebook.php]: one empty line should be before return;  please, fix :)

File: src/Codeception/Lib/Driver/Facebook.php (hunk 3 above)
Comment (enginvardar) [inline src/Codeception/Lib/Driver/Facebook.php]: Actually name is optional, if it's not set, default is 'Codeception Testuser' in module _initialize, i think defining test users name may help during tests.

Comment (DavertMik) [inline src/Codeception/Lib/Driver/Facebook.php]: If the name is optional then it should be passed as optional param as well, I think.

**Comment by DavertMik** [comment]
As for me this looks OK. @tiger-seo what about you? Should we merge or there are things that should be taken into account?

**Comment by tiger-seo** [comment]
it's ok ;)

**Comment by DavertMik** [comment]
Thanks! Merged

**Pull request merged**